openslides/utils/cache_providers.py

```diff
@@ -164,25 +170,59 @@ class RedisCacheProvider:
             local nc = tonumber(ARGV[1])
             local nd = tonumber(ARGV[2])
 
-            local i, max
-            -- Add changed_elements to the cache and sorted set (the first of the pairs)
+            local i, max, batch_counter
+            local change_id_data -- change_id, element_id, change_id, element_id, ...
+
+            -- Add changed_elements to the cache and sorted set using batches of 1000
+            -- values in unpack() (see #5386)
+            local elements -- element_id, element, element_id, element, ...
             if (nc > 0) then
-                max = 1 + nc
-                redis.call('hmset', KEYS[1], unpack(ARGV, 3, max + 1))
-                for i = 3, max, 2 do
-                    redis.call('zadd', KEYS[2], change_id, ARGV[i])
+                i = 3
+                max = 3 + nc
+                while (i < max) do
+                    change_id_data = {}
+                    elements = {}
+                    batch_counter = 1
+                    while (i < max and batch_counter <= 1000) do
+                        change_id_data[batch_counter] = change_id
+                        change_id_data[batch_counter + 1] = ARGV[i]
+                        elements[batch_counter] = ARGV[i]
+                        elements[batch_counter + 1] = ARGV[i + 1]
+                        batch_counter = batch_counter + 2
+                        i = i + 2
+                    end
+                    if (#change_id_data > 0) then -- so is #elements > 0
+                        redis.call('hmset', KEYS[1], unpack(elements))
+                        redis.call('zadd', KEYS[2], unpack(change_id_data))
+                    end
                 end
             end
 
             -- Delete deleted_element_ids and add them to sorted set
+            local element_ids -- element_id, element_id, ...
+            local element_ids_counter
             if (nd > 0) then
-                max = 2 + nc + nd
-                redis.call('hdel', KEYS[1], unpack(ARGV, 3 + nc, max))
-                for i = 3 + nc, max, 1 do
-                    redis.call('zadd', KEYS[2], change_id, ARGV[i])
+                i = 3 + nc
+                max = 3 + nc + nd
+                while (i < max) do
+                    change_id_data = {}
+                    element_ids = {}
+                    batch_counter = 1
+                    element_ids_counter = 1
+                    while (i < max and batch_counter <= 1000) do
+                        change_id_data[batch_counter] = change_id
+                        change_id_data[batch_counter + 1] = ARGV[i]
+                        element_ids[element_ids_counter] = ARGV[i]
+                        batch_counter = batch_counter + 2
+                        element_ids_counter = element_ids_counter + 1
+                        i = i + 2
```

Comment:
The `2` should be a `1`. We skipped every second elemnt in the cache while deleting it. This must have caused our problems.